### PR TITLE
Cross entropy loss and MultiFactorScheduler

### PR DESCRIPTION
**Pull request by taoari**
Calculate the cross entropy loss for softmax probability output. 

And also provide the MultiFactorScheduler that is able to set `step` to a list of integers to indicate where to reduce the learning rate. It expands the FactorScheduler, as the latter only accepts for a fixed `step`.

### CODE REVIEW

File: python/mxnet/lr_scheduler.py [hunk 1]
```diff
@@ -74,3 +74,52 @@ def __call__(self, num_update):
...
+    """
+    def __init__(self, step, factor=1):
+        super(MultiFactorScheduler, self).__init__()
+        assert isinstance(step, list) and len(step) >= 1
+        for i, _step in enumerate(step):
+            if i != 0 and step[i] <= step[i-1]:
+                raise ValueError("Schedule step must be an increasing integer list")
+            if _step < 1:
+                raise ValueError("Schedule step must be greater or equal than 1 round")
+        if factor >= 1.0:
```
Comment (piiswrong) [inline python/mxnet/lr_scheduler.py]: Consider change to >1.0. Currently the default factor will result in a error.

File: python/mxnet/lr_scheduler.py [hunk 2]
```diff
@@ -74,3 +74,52 @@ def __call__(self, num_update):
             logging.info("Update[%d]: Change learning rate to %0.5e",
                          num_update, self.base_lr)
         return self.base_lr
+
+class MultiFactorScheduler(LRScheduler):
```
Comment (piiswrong) [inline python/mxnet/lr_scheduler.py]: Consider merge with FactorScheduler and choose different behavior depending on step being a list or a number.

I don't have a very strong opinion on this so leave it as is if you think it's better.

File: python/mxnet/lr_scheduler.py (hunk 1 above)
Comment (taoari) [inline python/mxnet/lr_scheduler.py]: Which one, _step < 1 or len(step)>= 1? It seems that there is no problem on my local computer.

File: python/mxnet/lr_scheduler.py (hunk 2 above)
Comment (taoari) [inline python/mxnet/lr_scheduler.py]: I think leaving them separated is better. For example, I want to implement another Scheduler based FactorScheduler, it would not be so cumbersome to also consider the case of MultiFactorScheduler.

File: python/mxnet/lr_scheduler.py (hunk 1 above)
Comment (piiswrong) [inline python/mxnet/lr_scheduler.py]: if factor >= 1.0
default factor is 1 if you raise error when factor >= 1.0, the default value will cause error

Comment (taoari) [inline python/mxnet/lr_scheduler.py]: Oh yes. But in the default `FactorScheduler`, it is also using factor >= 1.0. I think the reason is that if factor is 1.0, one should set lr_scheduler to None, if `FactorScheduler` or `MultiFactorScheduler` is used, one should strictly set it < 1.0. Should I change the code to factor > 1.0 for both places?

Comment (piiswrong) [inline python/mxnet/lr_scheduler.py]: Change them both seem to make more sense since it's weird that default setting should fail.

Comment (taoari) [inline python/mxnet/lr_scheduler.py]: They are updated.

**Comment by piiswrong** [comment]
Thanks!
LGTM
